### PR TITLE
Use card component for token card

### PR DESCRIPTION
**Pull request by ncounter**
Based on top of https://github.com/openSUSE/open-build-service/pull/13809 to make use of the new default component in the token instances

A preview is available [here](https://obs-reviewlab.opensuse.org/ncounter-token-card/my/tokens)

## Before
token-card-before

## After
image

### CODE REVIEW

**Comment by ncounter** [comment]
[Unify Card Index Pattern (L)](https://trello.com/c/Ym4kvbUL/1872-unify-card-index-pattern-l)

File: src/api/app/components/token_card_component.html.haml [hunk 1]
```diff
@@ -1,37 +1,43 @@
...
+  - component.with_action do
+    - if policy(@token).webui_trigger?
+      = link_to(token_trigger_path(@token), title: 'Trigger Token', class: 'nav-link p-1') do
+        %i.fas.fa-project-diagram
+  - component.with_action do
+    - if @token.type == 'Token::Workflow'
+      = link_to(token_users_path(@token.id), title: 'Share Token', class: 'nav-link p-1') do
+        %i.fas.fa-users
+  - component.with_action do
+    = link_to(token_workflow_runs_path(@token.id), title: 'Workflow Runs', class: 'nav-link p-1') do
```
Comment (krauselukas) [inline src/api/app/components/token_card_component.html.haml:42]: @ncounter i think this also has to go under a `if @token.type == 'Token::Workflow'`

**Review by krauselukas — CHANGES_REQUESTED** [review]

File: src/api/app/components/token_card_component.html.haml (hunk 1 above)
Comment (krauselukas) [reply src/api/app/components/token_card_component.html.haml:42]: only available for workflow tokens

Comment (ncounter) [reply src/api/app/components/token_card_component.html.haml:42]: good catch, I probably overlooked it, thank you :+1

File: src/api/app/components/token_card_component.html.haml [hunk 2]
```diff
@@ -1,37 +1,43 @@
...
-          data: { 'bs-toggle': 'modal', 'bs-target': '#delete-token-modal', token_id: @token.id, action: token_path(@token) } do
-          %i.fas.fa-times-circle.text-danger
-        - if @token.type == 'Token::Workflow'
-          = link_to(token_users_path(@token.id), title: 'Share Token') do
-            %i.fas.fa-users
-          = link_to(token_workflow_runs_path(@token.id), title: 'Workflow Runs') do
-            %i.fas.fa-book-open
+
+  - component.with_delete_button do
+    = link_to '#', title: 'Delete Token',
```
[the remark this answers is not in the document]
Comment (ncounter) [reply src/api/app/components/token_card_component.html.haml]: that makes sense, done :+1:

**Comment by dmarcoux** [comment]
Whenever a token has an increased height due to being set for a project/package, other token cards look a bit weird due to the extra whitespace at the bottom. See here:

whitespace

The token description is also not shortened in the card header. The text should at least wrap on a second line.

File: src/api/app/views/webui/users/tokens/index.html.haml [hunk 3]
```diff
@@ -5,7 +5,7 @@
     %h3= @pagetitle
     .text-center.mb-2= page_entries_info(@tokens)
     - if @tokens.count.positive?
-      .d-flex.flex-wrap.gap-4
+      .d-flex.flex-wrap.gap-4.align-items-start
```
[the remark this answers is not in the document]
Comment (dmarcoux) [reply src/api/app/views/webui/users/tokens/index.html.haml]: What is the issue with having cards of slightly different sizes? By aligning actions at the bottom of the cards, we would end up with an empty area between the card's text content and its actions.

[the remark this answers is not in the document]
Comment (dmarcoux) [reply src/api/app/views/webui/users/tokens/index.html.haml]: > making the actions easier to find

It's true that consistency is good. I'll look into this.

[the remark this answers is not in the document]
Comment (dmarcoux) [reply src/api/app/views/webui/users/tokens/index.html.haml]: I've pushed changes for that. Please have a look :slightly_smiling_face:

**Review by krauselukas — COMMENTED** [review]
LGTM, after squashing the last commit I would click the approval button :)

**Comment by danidoni** [comment]
> LGTM, after squashing the last commit I would click the approval button :)

Squashed. Also, we can give it a :heavy_check_mark: and use the "Squash and merge" button at the bottom :grin:

**Review by krauselukas — APPROVED** [review]

**Pull request merged**